### PR TITLE
ESLint upgrade: Client and common packages use latest version of ESLint and shared config

### PR DESCRIPTION
**Pull request by tylerbutler**
1. Updates all **common and client packages** to use the latest shared ESLint config (0.25). All packages use the default `minimal` config. Future PRs can switch to a stricter config on a package-by-package basis.
2. Upgrades to ESLint 8.6.0, and upgrades plugins to compatible versions.
3. Removes eslint-plugin-prefer-arrow.
4. Updates fix script to not auto-fix unused disable comments. There are a number of false positives for some rules, (e.g. `prefer-read-only`) and auto removing the disable comments causes breaks when it shouldn't.
5. Updates the babel eslint parser used by some of the PropertyDDS projects to an ESLint 8-compatible version. This required some configuration changes.
6. Various code changes to address fallout from the upgrades.

### CODE REVIEW

File: packages/framework/undo-redo/src/sequenceHandler.ts [hunk 1]
```diff
@@ -124,6 +124,7 @@ export class SharedSegmentSequenceRevertible implements IRevertible {
                                     tracked.propertyDelta,
                                     undefined);
                             }
+                        // fallthrough
```
Comment (markfields) [inline packages/framework/undo-redo/src/sequenceHandler.ts:127]: This looks like it found a bug, like there should be a `break` statement above...?  @anthony-murphy, @wes-carlson what do you think?

File: packages/tools/merge-tree-client-replay/src/clientReplayTool.ts [hunk 2]
```diff
@@ -342,6 +342,7 @@ export class ClientReplayTool {
                                         }
                                     }
                                 }
+                            // fallthrough
```
Comment (markfields) [inline packages/tools/merge-tree-client-replay/src/clientReplayTool.ts]: optional nit: I'd rather see this fixed by putting `break;` in the previous case.

File: packages/tools/fetch-tool/src/fluidFetchArgs.ts [hunk 3]
```diff
@@ -121,6 +121,7 @@ export function parseArguments() {
             case "--help":
                 printUsage();
                 process.exit(0);
+            // fallthrough
```
Comment (markfields) [inline packages/tools/fetch-tool/src/fluidFetchArgs.ts]: optional nit: I'd rather see this fixed by putting `break;` in the previous case.

**Review by markfields — APPROVED** [review]
🎂 Thank you for making these difficult changes!  I only looked at the ts files, they LGTM.

File: packages/framework/undo-redo/src/sequenceHandler.ts (hunk 1 above)
Comment (markfields) [reply packages/framework/undo-redo/src/sequenceHandler.ts:127]: (If so, that change should come in a different PR)

File: packages/tools/merge-tree-client-replay/src/clientReplayTool.ts (hunk 2 above)
Comment (tylerbutler) [reply packages/tools/merge-tree-client-replay/src/clientReplayTool.ts]: Agreed, but I was nervous about changing the logic in some way. Adding the comment seemed safer (for me :)). Thanks for poking on this -- they could very well be bugs.

**Pull request merged**